test/e2e/common/networking.go

```diff
@@ -31,56 +31,52 @@ var _ = ginkgo.Describe("[sig-network] Networking", func() {
-			config := framework.NewCoreNetworkingTestConfig(f, true)
+		framework.ConformanceIt("should function for intra-pod communication: udp [NodeConformance]", func() {
+			config := framework.NewCoreNetworkingTestConfig(f, false)
 			for _, endpointPod := range config.EndpointPods {
 				config.DialFromTestContainer("udp", endpointPod.Status.PodIP, framework.EndpointUDPPort, config.MaxTries, 0, sets.NewString(endpointPod.Name))
 			}
 		})
 
 		/*
-			Release : v1.9
+			Release : v1.9, v1.16
 			Testname: Networking, intra pod http, from node
 			Description: Create a hostexec pod that is capable of curl to netcat commands. Create a test Pod that will act as a webserver front end exposing ports 8080 for tcp and 8081 for udp. The netserver service proxies are created on specified number of nodes.
 			The kubectl exec on the webserver container MUST reach a http port on the each of service proxy endpoints in the cluster using a http post(protocol=tcp)  and the request MUST be successful. Container will execute curl command to reach the service port within specified max retry limit and MUST result in reporting unique hostnames.
-			This test is marked LinuxOnly since HostNetwork is not supported on other platforms like Windows.
 		*/
-		framework.ConformanceIt("should function for node-pod communication: http [LinuxOnly] [NodeConformance]", func() {
-			config := framework.NewCoreNetworkingTestConfig(f, true)
+		framework.ConformanceIt("should function for node-pod communication: http [NodeConformance]", func() {
+			config := framework.NewCoreNetworkingTestConfig(f, false)
 			for _, endpointPod := range config.EndpointPods {
 				config.DialFromNode("http", endpointPod.Status.PodIP, framework.EndpointHTTPPort, config.MaxTries, 0, sets.NewString(endpointPod.Name))
 			}
 		})
 
 		/*
-			Release : v1.9
+			Release : v1.9, v1.16
 			Testname: Networking, intra pod http, from node
 			Description: Create a hostexec pod that is capable of curl to netcat commands. Create a test Pod that will act as a webserver front end exposing ports 8080 for tcp and 8081 for udp. The netserver service proxies are created on specified number of nodes.
 			The kubectl exec on the webserver container MUST reach a http port on the each of service proxy endpoints in the cluster using a http post(protocol=udp)  and the request MUST be successful. Container will execute curl command to reach the service port within specified max retry limit and MUST result in reporting unique hostnames.
-			This test is marked LinuxOnly since HostNetwork is not supported on other platforms like Windows.
 		*/
-		framework.ConformanceIt("should function for node-pod communication: udp [LinuxOnly] [NodeConformance]", func() {
-			config := framework.NewCoreNetworkingTestConfig(f, true)
+		framework.ConformanceIt("should function for node-pod communication: udp [NodeConformance]", func() {
```

Comment:
I'm not going to approve this change unless someone can explain:
1. What this test intends to verify
2. How it is doing that
3. Why changing this does not change the meaning of the test

Reply:
@bclau from my perspective we are still blocked on this, can you answer these questions?

Reply:
So, basically, there are 2 types of tests here:

- intra-pod communication
- node-pod communication

For both types of tests, these pods are spawned:

- 1 x netexec pod (``/agnhost netexec``) per node - named "EndpointPods"
- 1 x webserver pod (``/agnhost netexec``) - named "TestContainerPod"
- 1 x exec pod (``/agnhost pause``) - named "HostTestContainerPod" - in both cases, kubectl exec will be performed here.

So, basically, what happens in the tests:

- for each pod in EndpointPods, check the connectivity to that endpointPod through:
  - TestContainerPod for intra-pod communication tests, by curl-ing into it from the HostTestContainerPod (``http://%s/dial?request=hostName&protocol=%s&host=%s&port=%d&tries=1'`` - the ``netexec`` will dial into the given endpoint). This ensured that the communication was intra-pod, since both the TestContaienerPod and EndpointPod didn't have HostNetwork.
  - HostTestContainerPod for node-port communication tests - nc / curl directly into the EndpointPod. Since the test actually tests node's network <-> pod network connectivity, then it makes sense why this has HostNetwork.

Basically, the tests check for 2 types of network connectivity: pod <-> pod and node <-> pod.

The tests which are using ``DialFromNode`` no longer sets ``HostNetwork=false``, which won't change the those tests and what they verify, while ``HostNetwork`` is set to true for tests that do not use ``DialFromNode`` and only require pod-level network connectivity, allowing them to run on Windows too.